source/github-helpers/toast.tsx

```diff
@@ -12,14 +12,14 @@ export function ToastSpinner(): JSX.Element {
 }
 
 type ProgressCallback = (message: string) => void;
-type Task = (progress?: ProgressCallback) => Promise<unknown>;
-export default async function showToast<TTask extends Task>(
-	task: TTask,
+type Task<T> = (progress?: ProgressCallback) => Promise<T>;
+export default async function showToast<TaskOrError extends Task<any> | Error>(
+	task: TaskOrError,
```

Comment:
I think it should look like this:

```suggestion
type Task = (progress?: ProgressCallback) => Promise<unknown>;
export default async function showToast<TTask extends Task>(
	task: TTask | Error,
```

And then the return `void` as you mentioned.